### ISSUE TITLE
Mis-match between interaction and  interaction-handler

### ISSUE THREAD

**Issue by danielpeintner** [w3c/wot-scripting-api#188]
The current API has the following signature (e.g., for actions)
```
Promise<any> invokeAction(
      DOMString actionName,
      optional any params = null,
      optional InteractionOptions options = null
  );
```

Besides the name of action 2 more parameters are given (`params` and `options`).

The  `ActionHandler` has the following signature

`callback ActionHandler = Promise<any>(any parameters);`

One input parameter (`parameters`) only is expected.

Hence, a scripting runtime would need to combine `params` and `options` into one `parameters`. This seems undesirable. A *possible* fix would be to change the action handler to

`callback ActionHandler = Promise<any>(any parameters, InteractionOptions options);`

Note-1: the same applies for  `PropertyReadHandler` and `PropertyWriteHandler`
Note-2 (very minor): the names of the arguments could be harmonized so that it is clear that `parameters` in the handler is the same as `params` in invokeAction etc

**Comment by zolkis** [comment]
I guess we forgot to add the InteractionOptions to the handlers as well.
With Actions is a bit weird to have normal parameters and then other parameters exposed as URI variables, but it comes from TD design...

**Comment by zolkis** [comment]
Fixed by #189
